packages/editor/src/plugins/text/hooks/use-editable-paste-handler.tsx

```diff
@@ -47,6 +51,29 @@ export const useEditablePasteHandler = (args: UseEditablePasteHandlerArgs) => {
         }
       }
 
+      // pasting editor document string and insert as plugins
+      if (
+        shouldUseFeature('editorPluginCopyTool') &&
+        !media &&
```

Comment:
This line alone makes me think that it's time to split this hook into smaller functions, I'll try to think of a clean way to do it.

Reply:
this is my main concern with this PR right now (but also with other code): Where does stuff like this belong?
It's part of the text plugin, but also related to the rows plugin, could be a standalone plugin, could be a helper …

Reply:
I'd keep using this hook, but just extract the actual handling of different things into separate handler functions (which can still be in this same file). Then the root function (namely, the hook) only serves as a kind of switch statement, determining which handler needs to be called.